example/gluon/urban_sounds/train.py

```diff
@@ -0,0 +1,166 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+    The module to run training on the Urban sounds dataset
+"""
+import os
+import time
+import warnings
+import mxnet as mx
+from mxnet import gluon, nd, autograd
+from mxnet.gluon.contrib.data.audio.datasets import AudioFolderDataset
+from mxnet.gluon.contrib.data.audio.transforms import MFCC
+import model
+# Defining a function to evaluate accuracy
+def evaluate_accuracy(data_iterator, net):
+    acc = mx.metric.Accuracy()
+    for _, (data, label) in enumerate(data_iterator):
+        output = net(data)
+        predictions = nd.argmax(output, axis=1)
+        predictions = predictions.reshape((-1, 1))
+        acc.update(preds=predictions, labels=label)
+    return acc.get()[1]
+
+
+def train(train_dir=None, train_csv=None, epochs=30, batch_size=32):
+    """
+        The function responsible for running the training the model.
+    """
+    try:
+        import librosa
+    except ImportError:
+        warnings.warn("The dependency librosa is not installed. Cannot continue")
+        return
+    if not train_dir or not os.path.exists(train_dir) or not train_csv:
+        warnings.warn("No train directory could be found ")
+        return
+    # Make a dataset from the local folder containing Audio data
+    print("\nMaking an Audio Dataset...\n")
+    tick = time.time()
+    aud_dataset = AudioFolderDataset(train_dir, has_csv=True, train_csv=train_csv, file_format='.wav', skip_rows=1)
```

Comment:
1) True, I will remove the 'has_csv' keyword parameter.
2) The current design supports 2 ways in which it can read data:
    i) Like the ImageFolderDataset, it supports folder organization like ( label/<filename.wav> ),
    ii) second way is to have a folder(say Train) that contains all files and a csv file that has filename to label mapping. This can be extended for more fromats.